src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -488,6 +600,12 @@ public function addAllowedValues($option, $allowedValues)
             throw new AccessException('Allowed values cannot be added from a lazy option or normalizer.');
         }
 
+        if ($this->isNested($option)) {
+            @trigger_error(sprintf('The "%s" method should not be used with nested options. A failed attempt occurred with the option "%" for values %s', __METHOD__, $option, $this->formatValues($allowedValues)), E_USER_WARNING);
```

Comment:
That what I proposed in my [comment](https://github.com/HeahDude/symfony/commit/49588c6a053f717f5e14b383e235437a106c9bf8#commitcomment-16664140) in the concerned commit.

I'm not sure how strict we should be here, this is just a proposition.